frontend/src/metabase/visualizations/visualizations/Heading/Heading.tsx

```diff
@@ -39,32 +43,43 @@ export function Heading({
   settings,
   isEditing,
   isFullscreen,
-  isMobile,
   onUpdateVisualizationSettings,
 }: HeadingProps) {
   const inlineParameters = useSelector((state) =>
     getDashCardInlineValuePopulatedParameters(state, dashcard?.id),
   );
   const parameterValues = useSelector(getParameterValues);
-  const isEditingParameter = useSelector(getIsEditingParameter);
-
-  const mappingOptions = useSelector((state) =>
-    getDashcardParameterMappingOptions(state, {
-      card: dashcard.card,
-      dashcard,
-    }),
-  );
-  const hasVariables = mappingOptions.length > 0;
 
   const justAdded = useMemo(() => dashcard?.justAdded || false, [dashcard]);
 
-  const [isFocused, { turnOn: toggleFocusOn, turnOff: toggleFocusOff }] =
-    useToggle(justAdded);
+  const [isFocused, { open: toggleFocusOn, close: toggleFocusOff }] =
+    useDisclosure(justAdded);
   const isPreviewing = !isFocused;
 
   const [textValue, setTextValue] = useState(settings.text);
-  const preventDragging = (e: MouseEvent<HTMLInputElement>) =>
+  const preventDragging = (e: MouseEvent<HTMLInputElement>) => {
     e.stopPropagation();
+  };
+
+  const container = useRef<HTMLDivElement>(null);
+  const [isNarrow, setIsNarrow] = useState(false);
+
+  useEffect(() => {
+    const element = container.current;
+    if (!element) {
+      return;
+    }
+
+    const handleResize = (e: ResizeObserverEntry) => {
+      const { width } = e.contentRect;
+      setIsNarrow(width < 600); // Adjust the threshold as needed
+    };
+
+    resizeObserver.subscribe(element, handleResize);
+    return () => {
+      resizeObserver.unsubscribe(element, handleResize);
+    };
+  }, []);
```

Comment:
Thanks for adding that!! One small thing I noticed is that the `width` becomes `0` and edit mode and it seems like we're not remeasuring the code after saving changes, so filters stay in a menu until you refresh the page

https://github.com/user-attachments/assets/5c4431af-5112-40ef-aa55-46c530b2b650

Reply:
yeah I noticed that too, I think that's because the ref gets passed to two different components